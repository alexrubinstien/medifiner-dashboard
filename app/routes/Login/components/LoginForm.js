import React from 'react';
import {
  Field,
  reduxForm,
} from 'redux-form';
import Link from 'next/link';
import PropTypes from 'prop-types';

import Button from '@common/components/Button';
import ErrorComponent from '@common/components/ReduxFields/Error';
import Input from '@common/components/ReduxFields/Input';

import '@routes/Login/styles/LoginForm.scss';

const LoginForm = ({ handleSubmit }) => (
  <form
    className="login-form"
    onSubmit={handleSubmit}
  >
    <Field
      name="nonFieldErrors"
      component={ErrorComponent}
    />
    <div className="login-form__input-container">
      <Field
        name="email"
        type="email"
        placeholder="Email Address"
        component={Input}
      />
    </div>
    <div className="login-form__input-container">
      <Field
        name="password"
        type="password"
        placeholder="Password"
        component={Input}
      />
    </div>
    <p className="login-form__link-container">
      <Link href="/reset-password">
        <a className="login-form__reset-password-link">
          Reset your password
        </a>
      </Link>
    </p>
    <p className="login-form__button-container">
      <Button
        wide
        type="submit"
      >
        Login
      </Button>
    </p>
    <p className="login-form__text">
      Missing your login credentials? Please contact <a href="mailto:medfinder@healthmap.org">medfinder@healthmap.org</a>
    </p>
  </form>
);

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'login',
})(LoginForm);
