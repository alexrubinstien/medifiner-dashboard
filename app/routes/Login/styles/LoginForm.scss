.login-form {

  &__input-container + &__input-container {
    margin-top: 32px;
  }

  &__link-container {
    margin: 0;
    padding: 0;
  }

  &__reset-password-link {
    display: inline-block;
    font-size: 12px;
    font-weight: 500;
    line-height: 26px;
  }

  &__button-container {
    margin: 32px 0;
  }

  &__text {
    margin: 0;
  }
}
