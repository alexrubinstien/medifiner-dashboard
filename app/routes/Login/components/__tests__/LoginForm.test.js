/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import LoginForm from '../loginForm.js';

describe('With Snapshot Testing', () => {
  it('LoginForm shows fields with button', () => {
    const mockStore = configureStore([]);
    const store = mockStore();

    const component = renderer.create(
      <Provider store={store}>
        <LoginForm />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
