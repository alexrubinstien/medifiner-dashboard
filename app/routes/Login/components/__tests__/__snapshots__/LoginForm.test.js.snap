// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing LoginForm shows fields with button 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    className="error-component"
  />
  <div
    className="login-form__input-container"
  >
    <div
      className="field"
    >
      <div
        className="input input--wide"
      >
        <label
          className="input__label"
        />
        <input
          autoComplete="off"
          className="input__input field__input"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Email Address"
          type="email"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="login-form__input-container"
  >
    <div
      className="field"
    >
      <div
        className="input input--wide"
      >
        <label
          className="input__label"
        />
        <input
          autoComplete="off"
          className="input__input field__input"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <p
    className="login-form__link-container"
  >
    <a
      className="login-form__reset-password-link"
      href="/reset-password"
      onClick={[Function]}
    >
      Reset your password
    </a>
  </p>
  <p
    className="login-form__button-container"
  >
    <button
      className="button button--wide"
      type="submit"
    >
      Login
    </button>
  </p>
  <p
    className="login-form__text"
  >
    Missing your login credentials? Please contact 
    <a
      href="mailto:medfinder@healthmap.org"
    >
      medfinder@healthmap.org
    </a>
  </p>
</form>
`;
